handlers/callback.go

```diff
@@ -125,3 +128,127 @@ func HandleGovPayCallback(w http.ResponseWriter, req *http.Request) {
 	}
 	redirectUser(w, req, paymentSession.MetaData.RedirectURI, params)
 }
+
+func HandlePayPalCallback(externalPaymentSvc service.PaymentProviderService) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+		orderID := req.URL.Query().Get("token")
+		if orderID == "" {
+			log.ErrorR(req, fmt.Errorf("error orderID is blank"))
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		order, err := externalPaymentSvc.GetOrderDetails(orderID)
+		if err != nil {
+			log.ErrorR(req, fmt.Errorf("error getting order details: %v", err))
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		// The payment session must be retrieved directly to enable access to metadata outside the data block
+		paymentID := order.PurchaseUnits[0].ReferenceID
```

Comment:
We can get the payment id from the variable in the URL like [here](https://github.com/companieshouse/payments.api.ch.gov.uk/blob/master/handlers/callback.go#L21)